algorithms/src/polycommit/sonic_pc/mod.rs

```diff
@@ -432,15 +411,24 @@ impl<E: PairingEngine> PolynomialCommitment<E::Fr, E::Fq> for SonicKZG10<E> {
             labels.1.insert(label);
         }
 
-        assert_eq!(proof.len(), query_to_labels_map.len());
+        assert_eq!(proof.0.len(), query_to_labels_map.len());
 
         let mut randomizer = E::Fr::one();
 
-        let mut combined_comms: BTreeMap<Option<usize>, E::G1Projective> = BTreeMap::new();
-        let mut combined_witness: E::G1Projective = E::G1Projective::zero();
-        let mut combined_adjusted_witness: E::G1Projective = E::G1Projective::zero();
+        let mut combined_comms = BTreeMap::new();
+        let mut combined_witness = E::G1Projective::zero();
+        let mut combined_adjusted_witness = E::G1Projective::zero();
+
+        let mut batch_kzg_check_fs_rng = S::new();
+        batch_kzg_check_fs_rng.absorb_nonnative_field_elements(
```

Comment:
It seems like `absorb_nonnative_field_elements` only needs to iterate over its first argument (I checked the definition); it would avoid the vector allocations at the callsites

Reply:
The suggestion is to change the argument to `impl IntoIterator<Item = F>`? I think that makes sense.

Reply:
yes, that or `impl Iterator<Item = F>` or `I: Iterator<Item = F>`